File: functions/alsa-lib/variants/c/1_iec958_parity_8.c
#include <stdint.h>

static unsigned int iec958_parity(unsigned int data)
{
	data >>= 4;
	data ^= data >> 1;
	data ^= data >> 2;
	data ^= data >> 4;
	data ^= data >> 8;
	data ^= data >> 16;
	return data & 1;
}


int main(){
    iec958_parity(0);
}

File: functions/alsa-lib/variants/go/1_iec958_parity_8.go
package main

func iec958_parity(data uint32) uint32 {
	data >>= 4
	parity := uint32(0)
	for i := 4; i <= 30; i++ {
		if data&1 != 0 {
			parity++
		}
		data >>= 1
	}
	return parity & 1
}


func main(){
    iec958_parity(0)
}